README.md

```diff
@@ -1,6 +1,199 @@
 # plasmactl-bump
-Update the version of components that were updated in last commit
+Update the version of components that were updated in the last commit.
 
-### Excluded files
+### Excluded Files:
 - README.md
 - README.svg
+
+---
+
+### Work in Progress (WIP)
+
+**bump:** Updates the version of components that were modified after the last bump.  
+**Current bump logic:**
+1. Open the git repository.
+2. Check if the latest commit is not a bump commit.
+3. Collect a list of changed files until the previous bump commit is found (matching by author??). If the `last` option is passed, only take changes from the last commit. Prepare a map of resource objects to update.
+4. Get the short hash of the last commit.
+5. Iterate through the resource map and update the version of each.
+
+---
+
+### `bump --sync` or Propagation
+
+When resources are decoupled and stored in packages, there’s a need to preserve versions and maintain the current state of resources.
+`bump` works closely with `plasmactl compose`. The purpose of `compose` is to build the platform using resources from different places (domain repo, packages outside the repo).
+The purpose of `bump --sync` is to propagate the versions of a build to the current state.
+
+**How is it achieved?**
+
+#### Prerequisites:
+- It is important to use `bump --sync` after a fresh `compose`. Otherwise, an incorrect result or error may occur.
+
+### Overall Propagation Workflow:
+1. **Search and download the artifact to compare the build to:**
+  - Open the git repository.
+  - Search for the first bump commit after `HEAD`.
```

Comment:
- Search for the first bump commit after HEAD and use it's short sha (7 characters) to compute artifact name to retreive
- If no artifact exist with that name, recursively look for earlier bump commits